Source/WebKitLegacy/haiku/API/WebPage.cpp

```diff
@@ -812,66 +812,6 @@ void BWebPage::internalPaint(BView* offscreenView,
 }
 
 
-void BWebPage::scroll(int xOffset, int yOffset, const BRect& rectToScroll,
-       const BRect& clipRect)
-{
-    if (!rectToScroll.IsValid() || !clipRect.IsValid()
-        || (xOffset == 0 && yOffset == 0) || !fWebView->LockLooper()) {
-        return;
-    }
-
-    BBitmap* bitmap = fWebView->OffscreenBitmap();
-    BView* offscreenView = fWebView->OffscreenView();
-
-    // Lock the offscreen bitmap while we still have the
-    // window locked. This cannot deadlock and makes sure
-    // the window is not deleting the offscreen view right
-    // after we unlock it and before locking the bitmap.
-    if (!bitmap->Lock()) {
-       fWebView->UnlockLooper();
-       return;
-    }
-    fWebView->UnlockLooper();
-
-    BRect clip = offscreenView->Bounds();
-    if (clipRect.IsValid())
-        clip = clip & clipRect;
-
-    BRect rectAtSrc = rectToScroll;
-    BRect rectAtDst = rectAtSrc.OffsetByCopy(xOffset, yOffset);
-
-    // remember the part that will be clean
-    BRegion repaintRegion(rectAtSrc);
-    repaintRegion.Exclude(rectAtDst);
-    BRegion clipRegion(clip);
-    repaintRegion.IntersectWith(&clipRegion);
-
-    if (clip.Intersects(rectAtSrc) && clip.Intersects(rectAtDst)) {
-        // clip source rect
-        rectAtSrc = rectAtSrc & clip;
-        // clip dest rect
-        rectAtDst = rectAtDst & clip;
-
-        // move dest back over source and clip source to dest
-        rectAtDst.OffsetBy(-xOffset, -yOffset);
-        rectAtSrc = rectAtSrc & rectAtDst;
-        rectAtDst.OffsetBy(xOffset, yOffset);
-
-        offscreenView->CopyBits(rectAtSrc, rectAtDst);
```

Comment:
Removing this means the whole webpage is always redrawn everytime one scrolls, instead of just the newly exposed part. Is that ok? Wouldn't it make scrolling on slow to draw webpages even worse?

Reply:
Yes, this is a trade-off. Tiled backing store could probably solve both issues, but I'm reluctant to put more work into WebKitLegacy API.

Reply:
Fair point.
I had TiledBackingStore working at some point but it tends to create huge bitmaps (2048x2048) and several of them, if we would go in that direction we should at least remove the current offscreen buffer.

Well, I guess that's good enough for WebKitLegacy and start to see about moving things to WebKit2 indeed.